src/Orleans.Transactions/DistributedTM/TransactionAgent.cs

```diff
@@ -18,21 +60,29 @@ internal class TransactionAgent : ITransactionAgent
         private readonly ILogger logger;
         private readonly Stopwatch stopwatch = Stopwatch.StartNew();
         private readonly CausalClock clock;
+        private readonly TransactionAgentStatistics statistics;
+        private readonly TransactionOverloadDetector overloadDetector;
 
-        public TransactionAgent(IClock clock, ILogger<TransactionAgent> logger)
+        public TransactionAgent(IClock clock, ILogger<TransactionAgent> logger, TransactionAgentStatistics statistics, TransactionOverloadDetector overloadDetector)
```

Comment:
TransactionAgent should only reference the abstraction, not the concrete class.

Reply:
will fix this